src/modules/formBuilder/components/itemEditor/FormEditorItemProperties.tsx

```diff
@@ -77,6 +83,29 @@ const FormEditorItemProperties: React.FC<FormEditorItemPropertiesProps> = ({
     }));
   }, [itemTypeValue]);
 
+  const itemMap = useMemo(
+    () => getItemMap(definition.definition),
+    [definition.definition]
+  );
+
+  const onLabelBlur = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      // When the user edits the label, update the link ID to be a human-readable 'slugified' version of the label.
+      // But, don't make an update to the link ID if it has been manually set.
+      if (!dirtyFields.linkId) {
+        let newLinkId = slugifyItemLabel(event.target.value);
```

Comment:
@martha If I clear an existing label, it clears the Link ID. We probably don't want that. Maybe only if `!dirtyFields.linkId && event.target.value`

Reply:
Oh true, thanks!